src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -696,8 +696,129 @@ private Task WriteSuffix()
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe void EmitHttp11(ref MemoryPoolIterator end)
+        {
+            byte* b;
+            
+            if (end.GetRawBuffer(9, out b))
+            {
+                 // Emit directly
+                *(long*)b = 0x312E312F50545448;
```

Comment:
@JeffCyr. In what sense is this undefined? Are you worried about big endian architectures? Is the concern about misaligned writes simply a perf issue? In that case, wouldn't `Buffer.BlockCopy` also take a hit due to the misalignment?

I think there's the issue in the generating code where sometimes in the last assignment before `UpdateEnd` it's casting `b` to a `short*` when it should remain a `byte*`, causing the assignment to write one byte past the `End` index. In practice this isn't an issue since there is padding at the end of each block, but we would fix that before merging ofc.